scripts/sql/postgre/apollo_init_user.sql

```diff
@@ -0,0 +1,49 @@
+--
+-- Copyright 2023 Apollo Authors
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+-- http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+/* # CREATE apolloconfigdb; */
+/*  # ------------------------------------------------------------ */
+
+CREATE DATABASE "ApolloConfigDB";
+
+\c "ApolloConfigDB"
+
+CREATE SCHEMA apolloconfig;
+
+GRANT ALL PRIVILEGES ON DATABASE "ApolloConfigDB" to postgres;
+
+GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA apolloconfig to postgres;
+
+GRANT ALL PRIVILEGES ON ALL FUNCTIONS IN SCHEMA apolloconfig to postgres;
+
+GRANT ALL PRIVILEGES ON SCHEMA apolloconfig TO postgres;
+
+
+/* # CREATE apolloconfigdb; */
+/* # ------------------------------------------------------------ */
+
+CREATE DATABASE "ApolloPortalDB";
```

Comment:
It looks like Postgres is very different from MySQL, which would make future contributions harder.
I'm thinking how about we abstract a storage layer this time and move the Postgres support to a new git repository, e.g. apollo-storage-extensions? The apollo-storage-extensions repository would target a specific version of apollo and doesn't have to be updated every time a feature is added. It would instead be updated by some Postgres user from time to time when they think is necessary.

cc @apolloconfig/committers what do you think?

Reply:
I humbly suggest that we create a separate folder for now, as we are not entirely clear on which boundaries should be placed between multiple databases in the main code and extensions. My recommendation is that we start by placing PostgreSQL and other databases in a separate folder. Once we have successfully achieved compatibility with one or two databases, we can then begin to separate the repository.
cc @apolloconfig/committers